woocommerce/pinterest-for-woocommerce
# Add tracking for verify domain failed

### Changes proposed in this Pull Request:

Partially Closes #232.

domain_verify_failure | When a site verification fails  | step: the step of the process that failed (token, meta-tag, unknown)
-- | -- | -- 

### Screenshots:

![Screenshot 2021-12-14 at 16 06 16](https://user-images.githubusercontent.com/5908855/146024787-4ddc1928-8bb4-419a-a50d-bedc4b6fc177.png)
![Screenshot 2021-12-14 at 16 07 06](https://user-images.githubusercontent.com/5908855/146024792-f9b3bb21-9a96-4b7d-a6ed-ece70458d030.png)
![Screenshot 2021-12-14 at 16 07 50](https://user-images.githubusercontent.com/5908855/146024794-e8612c24-8767-4e3b-96a5-14e7ff85180d.png)

### Detailed test instructions:

0. `npm run dev` 
1. Go to https://38d2-213-94-22-45.ngrok.io/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding&step=claim-website in the onboarding page. If necessary, disconnect your previous account. 
2. Open the console and run `localStorage.setItem('debug', 'wc-admin*')` (be sure you have verbose mode activated) this is for activating the debug

*Testing failing in token step*

1. Go to `src/API/Base.php` and in line 154 remove ` . self::$token['access_token']`. 
2. If you click on the verification button you can see the track event `domain_verify_failure` with the `step` prop as `token`

*Testing failing in domain step*

1. Restore `src/API/Base.php`
2. Go to `src/API/DomainVerification.php` and in line 64 set the if condition as `false` to prevent the code to be executed and throw the exception regarding domain validation.
3. If you click on the verification button you can see the track event `domain_verify_failure` with the `step` prop as `domain verification`

*Testing failing in meta tag step*
1. Restore  `src/API/DomainVerification.php`
2. Go to `src/API/DomainVerification.php` and in line 70 set the if condition as `false` to prevent the code to be executed and throw the exception regarding meta tag validation.
3. If you click on the verification button you can see the track event `domain_verify_failure` with the `step` prop as `meta-tag`

### Changelog entry

> Tweak Added tracking in Domain Verification failure
